ConsumerApi.Tests.Integration/Features/Challenges/POST.feature

```diff
@@ -5,7 +5,7 @@ User creates a Challenge
 
 Scenario: Creating a Challenge as an anonymous user
 	Given the user is unauthenticated
-	When a POST request is sent to the Challenges endpoint
+	When a POST request is sent to the Challenges endpoint without authentication
```

Comment:
Because the SDK has `PostAuthenticated` and `PostUnauthenticated` method, which depending on the test need to be called.
Would you rather I change the logic of all endpoints to check whether the Client is authenticated?

Reply:
This is the only place where you added something like this. 

What's confusing me: I was just looking for a step definition for "When a POST request is sent to the Challenges endpoint without authentication", but I cannot find it anywhere. How is there no error in the tests? 😲  

What you should do though is: in the step definition for "When a POST request is sent to the Challenges endpoint", check for _isAuthenticated, and if it is `true`, call `sdk.Challenges.CreateChallenge()`. If it is `false`, call `sdk.Challenges.CreateChallengeUnauthenticated()`.